### PR TITLE
Make PR curves plugin work in SQL mode

### PR DESCRIPTION
**Pull request by chihuahua**
The plugin WAI:
image

### CODE REVIEW

File: tensorboard/plugins/pr_curve/pr_curves_plugin.py [hunk 1]
```diff
@@ -80,24 +83,75 @@ def pr_curves_impl(self, runs, tag):
...
-          for v in range(1, pr_curve_data.num_thresholds + 1)]
-
-      response_mapping[run] = [
-          self._process_tensor_event(e, thresholds) for e in tensor_events]
+    if self._db_connection_provider:
+      # Serve data from the database.
+      db = self._db_connection_provider()
+
+      # We select for steps greater than -1 because the writer inserts
+      # placeholder rows en masse. The check for step filters out those rows.
```
Comment (jart) [inline tensorboard/plugins/pr_curve/pr_curves_plugin.py]: Please do this using `dtype IS NOT NULL` instead, in accordance with the [schema definition](https://github.com/tensorflow/tensorflow/blob/4d9bef194799c0656c9ea80f8f083cfab1ebc14a/tensorflow/contrib/tensorboard/db/schema.cc#L92).

Comment (chihuahua) [reply tensorboard/plugins/pr_curve/pr_curves_plugin.py]: Thanks! Much more straightforward. Good to know the guarantees in the schema!

Comment (chihuahua) [reply tensorboard/plugins/pr_curve/pr_curves_plugin.py]: #1081

**Comment by jart** [comment]
> Originally, that was done to minimize the number of requests

A good way of thinking about this is, we want to eliminate superfluous requests; however, having more requests can be advantageous to promote parallelism. Please note TensorBoard should support HTTP/1.1 out of the box, so an individual TCP connection can serve many requests, without needing to be closed and reopened.

**Comment by chihuahua** [comment]
Ah yeah, if we get a performance boost from parallelism, that would be great! If indeed several XMLHttpRequests could map to 1 connection, then I think we can get around the max (I think [6ish](https://webmasters.stackexchange.com/questions/90301/how-can-i-tell-chrome-to-use-all-6-concurrent-connections-asap)) allowed connections at any time for Chrome.

**Comment by jart** [comment]
You could have a hundred XHRs and they'd just funnel into those 6ish allowed TCP sockets. The end result is still good, depending on the situation.

**Pull request merged**